src/Compilers/Core/Portable/CommandLine/AnalyzerConfigSet.cs

```diff
@@ -144,7 +156,8 @@ public AnalyzerConfigOptionsResult GetOptionsForSourcePath(string sourcePath)
                 TreeOptions.Builder treeBuilder,
                 AnalyzerOptions.Builder analyzerBuilder,
                 ArrayBuilder<Diagnostic> diagnosticBuilder,
-                string analyzerConfigPath)
+                string analyzerConfigPath,
+                SmallDictionary<ReadOnlyMemory<char>, string> diagIdCache)
```

Comment:
`diagnosticsIdCache`

Reply:
Why not `diagnosticsIdentifierCache`?

Reply:
Mostly the `diag` was bothering me. Fine with pretty much anything including the current.